ruby/ql/src/queries/meta/TaintedNodes.ql

```diff
@@ -0,0 +1,28 @@
+/**
+ * @name Tainted nodes
+ * @description Nodes reachable from a remote flow source via default taint-tracking steps.
+ * @kind problem
+ * @problem.severity recommendation
+ * @id rb/meta/tainted-nodes
+ * @tags meta
+ * @precision very-low
+ */
+
+import internal.TaintMetrics
+import codeql.ruby.DataFlow
+import codeql.ruby.TaintTracking
+
+class BasicTaintConfiguration extends TaintTracking::Configuration {
+  BasicTaintConfiguration() { this = "BasicTaintConfiguration" }
+
+  override predicate isSource(DataFlow::Node node) { node = relevantTaintSource(_) }
+
+  override predicate isSink(DataFlow::Node node) {
+    // To reduce noise from synthetic nodes, only count nodes that have an associated expression.
+    exists(node.asExpr().getExpr())
```

Comment:
So you will only report nodes that can be reached with an empty access path (i.e., where data is not stored into a field). Is that OK? If not, you would have to use partial flow, but I don't know how well that scales.

Reply:
> So you will only report nodes that can be reached with an empty access path (i.e., where data is not stored into a field). Is that OK?

Yes, that is on purpose.  
It's marking the nodes that would be flagged in a query, if the node was a sink.  

Sometimes adding a new step can make a node accessible with an empty access-path, where that same node was previously available with a non-empty access-path.  

And with partial flow I would have to set the exploration limit to (basically) infinite, and include the access-path in the alert-message. And I'm pretty sure that won't scale (although I haven't tested).